### ISSUE TITLE
CLRIE Message logging is too verbose

### ISSUE THREAD

**Issue by WilliamXieMSFT** [microsoft/CLRInstrumentationEngine#148]
Currently, the LogLevel Message causes extremely high overhead (to the point of uselessness for anything except development time, and even then it's quite large).

In addition, the current logging infrastructure could be refactored/restructured to delay dynamic allocations of strings for prefixes such as time from the InstrumentationMethod to some public interface supporting logging with VarArgs by the ProfilerManager. This would reduce memory size taken by CLRIE (and any snapshots generated) by reducing the memory page changes due to logging. This latter discussion could be moved to another Issue, but I want to jot it down somewhere for now.

**Comment by jakubch1** [comment]
Any plans on this?


Variable | Value | Description
-- | -- | --
MicrosoftInstrumentationEngine_LogLevel | "Errors\|Messages\|Dumps\|All" | Filters logs. Currently only "Errors" is allowed for EventLogging to prevent verbose logging.
MicrosoftInstrumentationEngine_LogLevel_GUID | "Errors\|Messages\|Dumps\|All" | Filters logs for InstrumentationMethods. Currently only "Errors" is allowed to prevent verbose logging.

Is this implemented? Can we set CLRIE to Errors and our instrumentation method to Dumps for example?

**Comment by WilliamXieMSFT** [comment]
Yes, details of how to use the logger can be found at https://github.com/microsoft/CLRInstrumentationEngine/blob/develop/docs/logging.md#logging-consumers-

Your InstrumentationMethod should obtain the logger via `IProfilerManager::GetLoggingInstance()`.
